### PR TITLE
fix: Round tax in description and add to shipping calculations (#21)

### PR DESCRIPTION
**Pull request by SilentHunter124**
* Tax Rate in Description field is now rounded
* Using description field on shipping fees and shipping fees taxes

### CODE REVIEW

File: ecommerce_integrations/shopify/order.py [hunk 1]
```diff
@@ -157,7 +158,7 @@ def get_order_taxes(shopify_order, setting):
 			{
 				"charge_type": _("On Net Total"),
 				"account_head": get_tax_account_head(tax),
-				"description": f"{get_tax_account_description(tax) or tax.get('title')} - {tax.get('rate') * 100.0}%",
+				"description": f"{get_tax_account_description(tax) or tax.get('title')} - {round(Decimal(tax.get('rate') * 100.0), 2)}%",
```
Comment (ankush) [inline ecommerce_integrations/shopify/order.py:161]:
```suggestion
				"description": f"{get_tax_account_description(tax) or tax.get('title')} - {tax.get('rate') * 100.0:.2f}%",
```


No need to use rounding for this. It's just for sake of showing the description and isn't actually used for any calculation here. You can just use the f-string format specifier here.

File: ecommerce_integrations/shopify/order.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 from frappe.utils import cstr, flt, get_datetime, getdate, nowdate, cint
 from shopify.collection import PaginatedIterator
 from shopify.resources import Order
+from decimal import Decimal
```
Comment (ankush) [inline ecommerce_integrations/shopify/order.py:8]:
```suggestion
```

File: ecommerce_integrations/shopify/order.py [hunk 3]
```diff
@@ -207,7 +208,7 @@ def update_taxes_with_shipping_lines(taxes, shipping_lines, setting):
 				{
 					"charge_type": _("Actual"),
 					"account_head": get_tax_account_head(shipping_charge),
-					"description": shipping_charge["title"],
+					"description": get_tax_account_description(shipping_charge) or shipping_charge["title"],
```
Comment (ankush) [inline ecommerce_integrations/shopify/order.py:211]: This is becoming a bit misleading now, anyway I'll refactor this later.

File: ecommerce_integrations/shopify/order.py [hunk 4]
```diff
@@ -218,7 +219,7 @@ def update_taxes_with_shipping_lines(taxes, shipping_lines, setting):
 				{
 					"charge_type": _("Actual"),
 					"account_head": get_tax_account_head(tax),
-					"description": tax["title"],
+					"description": f"{get_tax_account_description(tax) or tax.get('title')} - {round(Decimal(tax.get('rate') * 100.0), 2)}%",
```
Comment (ankush) [inline ecommerce_integrations/shopify/order.py:222]:
```suggestion
					"description": f"{get_tax_account_description(tax) or tax.get('title')} - {tax.get('rate') * 100.0:.2f}%",
```

**Review by ankush — CHANGES_REQUESTED** [review]
Thanks for fixing this. I think it's the multiplication with `100` that is causing this error. Anyhoo, I think it's better to use string formatter than actually round these values.

File: ecommerce_integrations/shopify/order.py (hunk 3 above)
Comment (SilentHunter124) [reply ecommerce_integrations/shopify/order.py:211]: All right, maybe this might be obsolete when using my other pullrequest...

**Review by ankush — APPROVED** [review]

**Pull request merged**